pennylane/ops/functions/bind_new_parameters.py

```diff
@@ -38,13 +38,51 @@ def bind_new_parameters(op: Operator, params: Sequence[TensorLike]) -> Operator:
 
     Args:
         op (.Operator): Operator to update
-        params (Sequence[TensorLike]): New parameters to create operator with
+        params (Sequence[TensorLike]): New parameters to create operator with. This
+            must have the same shape as `op.data`.
 
     Returns:
         .Operator: New operator with updated parameters
     """
+    try:
+        return op.__class__(*params, wires=op.wires, **copy.deepcopy(op.hyperparameters))
+    except TypeError:
```

Comment:
```suggestion
    except TypeError:
        # operation is doing something different with its call signature.
```

Reply:
> We could also just assume that templates are going to be weird, and always use the data setter for something located in the templates folder.

I can then get rid of the try-except block and instead have a "template" dispatch and an additional dispatch for `PauliRot` and `PCPhase`, which as far as I can tell are the only two "vanilla" operators which have a signature with a positional hyperparameter.

Reply:
this is fine, but we should track the improvement for it to be done sooner rather than later, because this would be quite costly once the function becomes more heavily used

Reply:
I'm going to run benchmarks on the gradients module PR once release stuff is over. I'll find other places where `bind_new_parameters` is being used and run benchmarks for them as well.